python/sdist/amici/petab/petab_importer.py

```diff
@@ -0,0 +1,2282 @@
+                    period0_assignments.setdefault(
+                        change.target_id, []
+                    ).append((exp_ind_id, change.target_value))
+            else:
+                self._create_period_start_event(
+                    experiment=experiment,
+                    i_period=i_period,
+                    period=period,
+                )
+
+        # Create initials for the first period
+        if period0_assignments:
+            free_symbols_in_assignments = set()
+            for target_id, changes in period0_assignments.items():
+                # The initial value might only be changed for a subset of
+                #  experiments. We need to keep the original initial value
+                #  for all other experiments.
+                target_entity = None
+                try:
+                    target_entity = next(
+                        c
+                        for c in model.components
+                        if c.name
+                        == self.map_petab_to_pysb.get(target_id, target_id)
+                    )
+                    if isinstance(target_entity, pysb.Parameter):
+                        default = target_entity.value
+                    elif isinstance(target_entity, pysb.Expression):
+                        default = target_entity.expr
+                    else:
+                        raise AssertionError(target_id)
+                except StopIteration:
+                    # species pattern?
+                    for initial in model.initials:
+                        if str(initial.pattern) == self.map_petab_to_pysb.get(
+                            target_id, target_id
+                        ):
+                            default = initial.value
+                            break
+                    else:
+                        raise AssertionError(target_id)
+
+                # Only create the initial assignment if there is
+                #  actually something to change.
+                if expr_cond_pairs := [
+                    (target_value, sp.Symbol(exp_ind) > 0.5)
+                    for exp_ind, target_value in changes
+                    if target_value != default
+                ]:
+                    # Unlike events, we can't have different initial
+                    #  assignments for different experiments, so we need to
+                    #  combine all changes into a single piecewise
+                    #  expression.
+                    expr = sp.Piecewise(
+                        *expr_cond_pairs,
+                        (default, True),
+                    )
+
+                    # Update the target expression
+                    if target_entity is not None and isinstance(
+                        target_entity, pysb.Expression
+                    ):
+                        target_entity.value = expr
+                    else:
+                        # if the target is not an expression, it must be an
```

Comment:
TODO: would be more efficient to convert this into a multi-model problem, but who knows whether we will ever get around to implementing that.

Reply:
Added as TODO.